pkg/controller/controllerutil/controllerutil.go

```diff
@@ -266,16 +238,7 @@ func validateOwner(owner, object metav1.Object) error {
 
 // Returns true if a and b point to the same object.
 func referSameObject(a, b metav1.OwnerReference) bool {
-	aGV, err := schema.ParseGroupVersion(a.APIVersion)
-	if err != nil {
-		return false
-	}
-
-	bGV, err := schema.ParseGroupVersion(b.APIVersion)
-	if err != nil {
-		return false
-	}
-	return aGV.Group == bGV.Group && a.Kind == b.Kind && a.Name == b.Name
+	return a.UID == b.UID
```

Comment:
/hold

No, this isn't correct as UIDs can change across backups and clusters. References should be compared like above

Reply:
Got it. I am gonna change this and I guess rest are okay

Reply:
And going to ensure if they are from the same namespace.

Reply:
Now this will ensure owner by validating namespace while removing owner ref or controller ref